utils/prepare_for_doc_test.py

```diff
@@ -92,6 +92,9 @@ def process_doc_file(code_file, add_new_line=True):
 
     # fmt: off
     splits = code.split("```")
+    if len(splits) % 2 != 1:
+        raise ValueError("The number of occurrences of ``` should be an even number.")
```

Comment:
Even number doesn't guarantee the correctness, but odd number definitely means something is wrong.

Reply:
(well, at least for almost all the cases in our model and doc files)